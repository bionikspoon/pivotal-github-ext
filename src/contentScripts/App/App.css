.App__container {
  margin: 15px 0 20px;
  padding-left: 60px;
}
