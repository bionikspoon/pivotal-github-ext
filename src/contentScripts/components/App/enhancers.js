/* @flow */
import { applySpec } from 'ramda'
import { withProps } from 'recompose'
import matchGroup from '../../utils/matchGroup'

type Props = {
  title: string,
}

export const storyIds = (props: Props): Array<string> => {
  const reStoryIds = /[^#]*#(\d+)(?=[,\]\s])/g

  return matchGroup(reStoryIds, 1, props.title)
}

export default withProps(applySpec({ storyIds }))
