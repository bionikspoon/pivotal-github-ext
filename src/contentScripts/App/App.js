import React from 'react'
import PropTypes from 'prop-types'
import PivotalStories from '../PivotalStories'
import './App.css'

export default function App(props) {
  return (
    <div className="App__container clearfix">
      <PivotalStories storyIds={props.storyIds} />
    </div>
  )
}

App.propTypes = {
  storyIds: PropTypes.arrayOf(PropTypes.node.isRequired).isRequired,
}
