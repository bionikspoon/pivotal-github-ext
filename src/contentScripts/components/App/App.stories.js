/* @flow */
import React from 'react'

import { storiesOf } from '@storybook/react'
import App from './'
import GithubWrapper, { StyleWrapper } from '../GithubWrapper'
import MockChrome from '../../utils/MockChrome/MockChrome'

storiesOf('App', module)
  .add('default', () => (
    <StyleWrapper>
      <App title="Makin' Changes [#151356728]" chrome={new MockChrome({})} />
    </StyleWrapper>
  ))
  .add('multiple stories', () => (
    <StyleWrapper>
      <App
        title="Makin' Changes [#151356728, #151371609, #151371625]"
        chrome={new MockChrome({})}
      />
    </StyleWrapper>
  ))
  .add('github', () => {
    const title = "Makin' Changes [#151356728]"

    return (
      <GithubWrapper title={title}>
        <App title={title} chrome={new MockChrome({})} />
      </GithubWrapper>
    )
  })
