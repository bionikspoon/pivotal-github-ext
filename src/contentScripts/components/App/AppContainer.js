/* @flow */
import * as React from 'react'
import './App.css'

type Props = { children: React.Node }

export default function AppContainer(props: Props) {
  return <div className="App__container clearfix">{props.children}</div>
}
