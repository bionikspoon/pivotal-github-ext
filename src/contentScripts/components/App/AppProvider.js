/* @flow */
import * as React from 'react'
import './App.css'
import ChromeContext from '../ChromeContext/ChromeContext'

type Props = { children: React.Node, chrome: Object }

export default function AppContainer(props: Props) {
  return (
    <ChromeContext.Provider value={props.chrome}>
      {props.children}
    </ChromeContext.Provider>
  )
}
