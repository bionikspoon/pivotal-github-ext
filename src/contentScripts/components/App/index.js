/* @flow */
import App from './App'
import enhancer from './enhancers'

export default enhancer(App)
