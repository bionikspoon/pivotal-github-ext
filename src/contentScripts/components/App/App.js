/* @flow */
import React from 'react'
import PivotalStories from '../PivotalStories'
import AppProvider from './AppProvider'

type Props = {
  storyIds: Array<string>,
  chrome?: Object,
}

export default function App(props: Props) {
  return (
    <AppProvider chrome={props.chrome}>
      <div className="App__container clearfix">
        <PivotalStories storyIds={props.storyIds} />
      </div>
    </AppProvider>
  )
}

App.defaultProps = {
  chrome: global.chrome,
}
