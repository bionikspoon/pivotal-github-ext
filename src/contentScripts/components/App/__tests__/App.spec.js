import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('components/App', () => {
  test('it renders correctly', () => {
    const props = {
      storyIds: ['1234', '5678'],
    }
    const wrapper = shallow(<App {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
