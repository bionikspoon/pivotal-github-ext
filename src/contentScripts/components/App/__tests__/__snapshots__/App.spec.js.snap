// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App it renders correctly 1`] = `
<AppContainer>
  <PivotalStories
    fetchResources={[Function]}
    storage={[Function]}
    storyIds={
      Array [
        "1234",
        "5678",
      ]
    }
  />
</AppContainer>
`;
